src/viur/core/bones/phone.py

```diff
@@ -0,0 +1,127 @@
+
+        Args:
+            test (t.Optional[t.Pattern[str]]): An optional custom regex pattern for phone number validation.
+            max_length (int): The maximum length of the phone number. Passed to 'StringBone'.
+            default_country_code (t.Optional[str]): The default country code to apply if none is provided.
+            **kwargs (t.Any): Additional keyword arguments. Passed to 'StringBone'.
+        """
+        self.test: t.Pattern[str] = re.compile(test) if isinstance(test, str) else test
+        self.default_country_code: str = default_country_code
+
+        super().__init__(max_length=max_length, **kwargs)
+
+    @staticmethod
+    def _extract_digits(value: str) -> str:
+        """
+        Extracts and returns only the digits from the given value.
+
+        Args:
+            value (str): The input string from which to extract digits.
+
+        Returns:
+            str: A string containing only the digits from the input value.
+        """
+        return re.sub(r"[^\d+]", "", value)
+
+    def isInvalid(self, value: str) -> t.Optional[str]:
+        """
+        Checks if the provided phone number is valid or not.
+
+        Args:
+            value (str): The phone number to be validated.
+
+        Returns:
+            t.Optional[str]: An error message if the phone number is invalid or None if it is valid.
+
+        The method checks if the provided phone number is valid according to the following criteria:
+        1. The phone number must not be empty.
+        2. The phone number must match the provided or default phone number format.
+        3. The phone number cannot exceed 15 digits, or the specified maximum length if provided (digits only).
+        """
+        if not value:
+            return "No value entered"
+
+        if not self.test.match(value):
+            return "Invalid phone number entered"
+
+        # make sure max_length is not exceeded.
+        is_invalid: t.Optional[str] = super().isInvalid(self._extract_digits(value))
+
+        if is_invalid:
+            return is_invalid
+        return None
+
+    def singleValueFromClient(
+        self, value: str, skel: t.Any, bone_name: str, client_data: t.Any
+    ) -> t.Tuple[t.Optional[str], t.Optional[t.List[ReadFromClientError]]]:
+        """
+        Processes a single value from the client, applying the default country code if necessary and validating the
+        phone number.
+
+        Args:
+            value (str): The phone number provided by the client.
+            skel (t.Any): Skeleton data (not used in this method).
+            bone_name (str): The name of the bone (not used in this method).
+            client_data (t.Any): Additional client data (not used in this method).
+
+        Returns:
+            t.Tuple[t.Optional[str], t.Optional[t.List[ReadFromClientError]]]:
+            A tuple containing the processed phone number and an optional list of errors.
```

Comment:
Please remove the types here, the type hints in the signature are enough.
And please use the [Sphinx docstring format](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format).